code/modules/surgery/organs/liver.dm

```diff
@@ -0,0 +1,68 @@
+#define LIVER_DEFAULT_HEALTH 100 //amount of damage required for liver failure
+#define LIVER_DEFAULT_TOX_TOLERANCE 3 //amount of toxins the liver can filter out
+#define LIVER_DEFAULT_TOX_LETHALITY 0.5 //lower values lower how harmful toxins are to the liver
+
+/obj/item/organ/liver
+	name = "liver"
+	icon_state = "liver"
+	origin_tech = "biotech=3"
+	w_class = WEIGHT_CLASS_NORMAL
+	zone = "chest"
+	slot = "liver"
+	desc = "Pairing suggestion: chianti and fava beans."
+	var/damage = 0 //liver damage, 0 is no damage, 100 causes liver failure
+	var/alcohol_tolerance = ALCOHOL_RATE
+	var/failing //is this liver failing?
+	var/maxHealth = LIVER_DEFAULT_HEALTH
+	var/toxTolerance = LIVER_DEFAULT_TOX_TOLERANCE
+	var/toxLethality = LIVER_DEFAULT_TOX_LETHALITY
+	var/filterToxins = TRUE //whether to filter toxins
+	var/list/toxinList
+
+/obj/item/organ/liver/New()
+	..()
+	toxinList = typesof(/datum/reagent/toxin)
+
+
+/obj/item/organ/liver/on_life()
+	var/mob/living/carbon/C = owner
+
+	//slowly heal liver damage
+	if(damage > 0)
+		damage -= 0.01//in most cases the pure toxin damage from too much alcohol will kill you first.
+	if(damage > maxHealth)//cap liver damage
+		damage = maxHealth
+
+	if(istype(C))
+		if(!failing)//can't process reagents with a failing liver
+			if(C.reagents)
+				if(filterToxins)
+					//handle liver toxin filtration
+					var/toxamount
+					for(var/toxin in toxinList)
+						var/datum/reagent/toxin/T = toxin
+						if(istype(T))
+							toxamount += C.reagents.get_reagent_amount(T.type)
+
+					if(toxamount <= toxTolerance && toxamount > 0)
+						C.reagents.remove_all_type("toxin", toxTolerance, 1)

```

Comment:
The first argument of remove_all_type seems to be a typepath, and not a chemical's string identifier.